logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -604,7 +605,7 @@ class TestPipeline < LogStash::Pipeline
 
       Thread.new { subject.run }
       # make sure we have received all the generated events
-      sleep 1 while dummyoutput.events.size < number_of_events
+      sleep 0.25 until dummyoutput.events.size == number_of_events
```

Comment:
Using exact equality is a less defensive test than `<`. If we somehow output more events than expected due to a bug we want this to terminate not go on infinitely, so that it can fail.